### ISSUE TITLE
Related resources cannot find security-group for every resource except ec2.

### ISSUE THREAD

**Issue by aiell0** [capitalone/cloud-custodian#541]
I have begun to use the related resource functionality that was just merged on 10/17, and I have noticed some issues with the usage. I am not sure if it is because of my lack of understanding of the syntax, or if the functionality is broken. I didn't see any testing uploaded with the merge so it is possible these issues were missed.

For EC2, I want to get rid of instances that are using certain security groups. I do not know if this syntax is the most efficient, but I have gotten it to work using the following template (security group ids have been taken out):

```
  - name: ec2-default-security-group
    description: |
      If EC2 is using default security group, terminate the instance.
    resource: ec2
    filters:
      - or:
        - type: security-group
          match_resource: true
          key: "GroupId"
          value: "sg-xxxxxxxx"
        - type: security-group
          match_resource: true
          key: "GroupId"
          value: "sg-xxxxxxxx"
        - type: security-group
          match_resource: true
          key: "GroupId"
          value: "sg-xxxxxxxx"
        - type: security-group
          match_resource: true
          key: "GroupId"
          value: "sg-xxxxxxxx"
    actions:
      - terminate
```

Now, I am trying to replicate this functionality among every applicable resource. Let's take redshift for example:

```
  - name: redshift-cluster-default-security-group
    description: |
      If Redshift Cluster is using default security group, terminate the instance.
    resource: redshift
    filters:
      - or:
        - type: security-group
          match_resource: true
          key: "VpcSecurityGroupId"
          value: "sg-xxxxxxxx"
        - type: security-group
          match_resource: true
          key: "VpcSecurityGroupId"
          value: "sg-xxxxxxxx"
        - type: security-group
          match_resource: true
          key: "VpcSecurityGroupId"
          value: "sg-xxxxxxxx"
        - type: security-group
          match_resource: true
          key: "VpcSecurityGroupId"
          value: "sg-xxxxxxxx"
    actions:
      - type: delete
        skip-snapshot: true
```

That policy breaks with the following errors:

```
2016-10-20 19:06:33,541: custodian.policy:INFO Running policy redshift-cluster-default-security-group resource: redshift region:us-east-1 c7n:0.8.19.6
2016-10-20 19:06:33,932: custodian.resources.securitygroup:WARNING event ids not resolved: ['sg-7ab5dd02'] error:An error occurred (InvalidGroup.NotFound) when calling the DescribeSecurityGroups operation: The security group 'sg-7ab5dd02' does not exist in default VPC 'vpc-701e2515'
2016-10-20 19:06:33,933: custodian.filters:WARNING Resource cluster:dbr-redshift-redshiftcluster-o5tiqoqftuv3 references non existant SecurityGroup: sg-7ab5dd02
2016-10-20 19:06:33,988: custodian.resources.securitygroup:WARNING event ids not resolved: ['sg-7ab5dd02'] error:An error occurred (InvalidGroup.NotFound) when calling the DescribeSecurityGroups operation: The security group 'sg-7ab5dd02' does not exist in default VPC 'vpc-701e2515'
2016-10-20 19:06:33,988: custodian.filters:WARNING Resource cluster:dbr-redshift-redshiftcluster-o5tiqoqftuv3 references non existant SecurityGroup: sg-7ab5dd02
2016-10-20 19:06:34,038: custodian.resources.securitygroup:WARNING event ids not resolved: ['sg-7ab5dd02'] error:An error occurred (InvalidGroup.NotFound) when calling the DescribeSecurityGroups operation: The security group 'sg-7ab5dd02' does not exist in default VPC 'vpc-701e2515'
2016-10-20 19:06:34,038: custodian.filters:WARNING Resource cluster:dbr-redshift-redshiftcluster-o5tiqoqftuv3 references non existant SecurityGroup: sg-7ab5dd02
2016-10-20 19:06:34,089: custodian.resources.securitygroup:WARNING event ids not resolved: ['sg-7ab5dd02'] error:An error occurred (InvalidGroup.NotFound) when calling the DescribeSecurityGroups operation: The security group 'sg-7ab5dd02' does not exist in default VPC 'vpc-701e2515'
2016-10-20 19:06:34,090: custodian.filters:WARNING Resource cluster:dbr-redshift-redshiftcluster-o5tiqoqftuv3 references non existant SecurityGroup: sg-7ab5dd02
2016-10-20 19:06:34,090: custodian.resources.redshift:INFO Filtered from 1 to 0 redshift
2016-10-20 19:06:34,090: custodian.policy:INFO policy: redshift-cluster-default-security-group resource:redshift has count:0 time:0.55
```

I'm not sure if my syntax is incorrect or if this is broken, but these are my results.

There is also the concern of what to do when the security groups come in as an array, like so:

```
  - name: elb-default-security-group
    description: |
      If elastic load balancer is using default security group, terminate the instance.
    resource: elb
    filters:
      - or:
        - type: security-group
          match_resource: true
          key: "SecurityGroups[]"
          value: "sg-xxxxxxxx"
        - type: security-group
          match_resource: true
          key: "SecurityGroups[]"
          value: "sg-xxxxxxxx"
        - type: security-group
          match_resource: true
          key: "SecurityGroups[]"
          value: "sg-xxxxxxxx"
        - type: security-group
          match_resource: true
          key: "SecurityGroups[]"
          value: "sg-xxxxxxxx"
    actions:
      - delete
```

How would the filters look in this case?

**Comment by kapilt** [comment]
sounds like a gap and we need some more unit tests on this for other resource types.

**Comment by skurfer** [comment]
It ends up doing a query for `GroupNames=['sg-xxxxxxxx']`. Still trying to track down why.

If you change the query to `GroupIds=['sg-xxxxxxxx']` or `GroupNames=['default']`, it works.

**Comment by skurfer** [comment]
I think there may be an upstream bug in skew. In their `SecurityGroup` the ID and filter don’t seem to match. (One refers to the group’s ID and one to its name.) This doesn’t look consistent with the other resources described in that file.

See https://github.com/scopely-devops/skew/blob/develop/skew/resources/aws/ec2.py#L44

I’ll open an issue there to confirm.

FYI, I was able to get the resources to show up by changing the code in skew, and by continuing to use `GroupId` as the key instead of `VpcSecurityGroupId`.

**Comment by kapilt** [comment]
definitely good to address upstream, but at this point i'd like to just get the fix in, its a one liner, reassigning, and addressed. we primarily use the skew models as defaults with extensions and mods as nesc, and use our own query engine as well.

wrt to the upstream, this is a delta i think from trying to support the classic mode networking, which has been deprecated for several years.

**Comment by kapilt** [comment]
addressed in #564

**Comment by nitinitare** [comment]
Hi @kapilt  @skurfer can you please share link for security group policies.
I Want to allow port 80 nad 443 to have cidr value 0.0.0.0/0. No other port should have 0.0.0.0/0. 
Please Provide the policy reference for that.
